core/src/agent/state.rs

```diff
@@ -134,20 +134,22 @@ fn reduce_commit_entry(
 
     // @TODO adding the entry to the CAS should happen elsewhere.
     fn response(
+        context: &Context,
         state: &mut AgentState,
         entry: &Entry,
         chain_header: &ChainHeader,
     ) -> Result<Address, HolochainError> {
-        state.chain.content_storage().add(entry)?;
-        state.chain.content_storage().add(chain_header)?;
+        let mut res_context = context.clone();
+        res_context.file_storage.add(entry)?;
+        res_context.file_storage.add(chain_header)?;
```

Comment:
Could it be refactored with the dht storage? I could revert it back if that is the case. I only added this because it was only an in memory store for persistence.

Reply:
Oh yeah Chainstore takes in a CAS so I will make that change definitely. Will probably cause an explosion of the state type all around but no problem